pkg/status/status.go

```diff
@@ -0,0 +1,156 @@
+package status
+
+import (
+	configv1 "github.com/openshift/api/config/v1"
+	configclient "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	cohelpers "github.com/openshift/library-go/pkg/config/clusteroperator/v1helpers"
+	operatorhelpers "github.com/openshift/library-go/pkg/operator/v1helpers"
+	"github.com/operator-framework/operator-marketplace/version"
+	log "github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const clusterOperatorName = "marketplace-operator"
+
+type status struct {
+	client          *configclient.ConfigV1Client
+	coAPINotPresent bool
+	namespace       string
+	clusterOperator *configv1.ClusterOperator
+}
+
+// Status is the interface to report the health of the operator to CVO
+type Status interface {
+	SetFailing(message string)
+	SetAvailable(message string)
+}
+
+// New returns an initialized Status
+func New(client *configclient.ConfigV1Client, namespace string) Status {
+	// Check if the ClusterOperator API is present on the cluster. This is so
+	// that we can continue to work with Kubernetes clusters that don't have
+	// this API.
+	_, err := client.ClusterOperators().Get(clusterOperatorName, v1.GetOptions{})
+	coAPINotPresent := false
+	if meta.IsNoMatchError(err) {
```

Comment:
@kevinrizza @tkashem I discovered that the error returned when `ClusterOperator` CRD is not present is a generic not found error. This won't help us on vanilla kube. So instead I am going to push a change where I look for the `ClusterOperator` CRD itself. The downside is that I need to add the CRD scheme to the manager. Let me know what you think after I push.